mam/v2/api/api.c

```diff
@@ -52,7 +57,10 @@ retcode_t mam_api_destroy(mam_api_t *const api) {
   mam_ntru_sk_t_set_free(&api->ntru_sks);
   mam_ntru_pk_t_set_free(&api->ntru_pks);
   mam_psk_t_set_free(&api->psks);
-  ret = trit_t_to_mam_msg_send_context_t_map_free(&api->send_ctxs);
+  ERR_BIND_RETURN(trit_t_to_mam_msg_send_context_t_map_free(&api->send_ctxs),
+                  ret);
+  ERR_BIND_RETURN(trit_t_to_mam_msg_recv_context_t_map_free(&api->recv_ctxs),
+                  ret);
```

Comment:
We can't return if one fail, it would mean that we doesn't free the others.
Also, our usage `ERR_BIND_RETURN` across MAM is very inconsistent, not sure what to do with that.

Reply:
true